### ISSUE TITLE
Enforce RegEx pattern in comment

### ISSUE THREAD

**Issue by blackswanny** [eslint/eslint#11421]
**Please describe what the rule should do:**

Enforce RegEx pattern in comment. So some substring of the comment should be present in comment always.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[x] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
/* JIRA-12345 some comment */
/* REF:123 some other comment */
/* some comment LINK-URL */
```

**Why should this rule be included in ESLint (instead of a plugin)?**
This is common case to refer to original documentation or any other source in comments as well as have some rules to be used to have code of conduct for comment in particular project

**Are you willing to submit a pull request to implement this rule?**
Yes

**Comment by not-an-aardvark** [comment]
Hi, thanks for creating an issue.

To clarify, do you mean that you want every single comment in your code to have a substring like JIRA-12345? It seems like most codebases would have at least some comments that are just describing why/how code does something, without linking to a jira ticket.

**Comment by blackswanny** [comment]
in my observation there are mostly 3 types of teams:
1) those who hate comments at all and prefer self-descriptive code and knowledge to be saved elsewhere
2) those who have any comments with no notation or rules (or may be JSDoc rules)
3) those who have rules for comments (if any is added to code) to still link comment to some broader knowledge base and not be as chaotic as 2) option

So 3) appears to be compromise between 1) and 2). And this eslint rule is mostly for this group. So I anticipate that random comments may still appear in such code, but mostly as exception, so in this case they can always use rare ` eslint-disable-next-line`

**Comment by not-an-aardvark** [comment]
I see, thanks for explaining. 

Personally, I'm not sure this rule would make sense to add to ESLint core -- it seems like a fairly specific stylistic preference, since I think most codebases do occasionally use comments without links. We wouldn't be able to add individual rules for every user's preference, so instead we generally only add a rule if it would be useful for a fairly large percentage of our users, and in other cases we suggest that people [create a custom rule](https://eslint.org/docs/developer-guide/working-with-rules) instead. Personally, I'm not convinced this rule would meet the bar for adding to ESLint core, although I wouldn't object to adding it if another team member thinks it's worthwhile.

**Comment by blackswanny** [comment]
of course for me it doesn't look very specific. There are similar rules in eslint core, which already look specific. They don't follow any standards of JS, W3C, common styleguides or etc. Like those below:
https://eslint.org/docs/rules/capitalized-comments
https://eslint.org/docs/rules/no-warning-comments

Wanted to confirm the way to add custom rule:
1) Create custom rule file `somerule.js`
2) in `eslint.rc` setup this rule , where `ruleId` equals file name (without `.js`)
3) add `--rulesdir` in CLI

correct?
can I add `rulesdir` as a part of config? can I not provide `rulesdir` if the file is put in the same dir?

**Comment by not-an-aardvark** [comment]
> Wanted to confirm the way to add custom rule:

Yes, that's correct. The `--rulesdir` file currently needs to be on the command line, and can't be specified in config. Alternatively, you could create a [plugin](https://eslint.org/docs/developer-guide/working-with-plugins) and put the rule in it, then add the plugin as a `devDependency`.
